File: Java/1525. Number of Good Ways to Split a String.java
class Solution {
    private int counter(int t[]){
        int res = 0;
        for(int i: t) res += i == 0 ? 0 : 1;
        return res;
    }

    public int numSplits(String s) {
        int l[] = new int[26], r[] = new int[26], ans = 0, temp;

        for(char i: s.toCharArray()) r[(int)i - 97]++;

        for(char i: s.toCharArray()){
            temp = (int)i - 97;
            if(counter(l) == counter(r)) ans++;
            l[temp]++;
            r[temp]--;
        }

        return ans;
    }
}

File: Python/1525. Number of Good Ways to Split a String.py
class Solution:
    def length(self, counter) -> int:
        return len([x for x, i in counter.items() if i > 0])

    def numSplits(self, s: str) -> int:
        l, r, res = Counter(), Counter(s), 0

        for i in s:
            if self.length(l) == self.length(r): res += 1
            l[i] += 1
            r[i] -= 1
        
        return res
